### PR TITLE
Adding my_hwm_package

### PR DESCRIPTION
**Pull request by earthai-tech**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by earthai-tech** [comment]
@conda-forge/help-python

**Comment by earthai-tech** [comment]
@conda-forge/help-python

File: recipes/hwm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 5f09159edf703390f4d3110d4737b2f2c47754a8bf11c2b0993e67270773f8c8
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:15]: Please use grayskull to generate Python recipes. Grayskull make it easier for us to review.

See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi for more information.

**Comment by earthai-tech** [comment]
@conda-forge/help-python  
Hello, I am reaching out to understand why, despite multiple attempts, building from OSX continues to fail. Any insights or suggestions would be greatly appreciated.

Thank you!

**Comment by ocefpaf** [comment]
> @conda-forge/help-python Hello, I am reaching out to understand why, despite multiple attempts, building from OSX continues to fail. Any insights or suggestions would be greatly appreciated.

See https://github.com/conda-forge/staged-recipes/issues/28242

File: recipes/hwm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    - hwm
+    - hwm.api
+    - hwm.compat
+    - hwm.datasets
+    - hwm.estimators
+    - hwm.utils
+  commands:
+    - python -c "import hwm; print(hwm.__version__)"
+  requires:
+    - python >=3.9
```
Comment (synapticarbors) [inline recipes/hwm/meta.yaml:43]: Please follow the convention detailed in:
https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python

for setting `python_min` and how they are applied in each of the requirements sections.

**Review by synapticarbors — CHANGES_REQUESTED** [review]

**Comment by earthai-tech** [comment]
@conda-forge/help-python

**Comment by synapticarbors** [comment]
You still haven't addressed my request from 3 weeks ago: https://github.com/conda-forge/staged-recipes/pull/28204#pullrequestreview-2443418982

File: recipes/hwm/meta.yaml (hunk 2 above)
Comment (earthai-tech) [reply recipes/hwm/meta.yaml:43]: I fix it :
```yaml
{% set name = "hwm" %}
{% set version = "1.0.1" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/hwm-{{ version }}.tar.gz
  sha256: 5f09159edf703390f4d3110d4737b2f2c47754a8bf11c2b0993e67270773f8c8

build:
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
  number: 0

requirements:
  host:
    - python >=3.9
    - pip
  run:
    - python >=3.9
    - numpy <2
    - scipy >=1.7.0
    - scikit-learn >=1.2.0
    - pandas >=1.3.0

test:
  imports:
    - hwm
  commands:
    - pip check
  requires:
    - pip

about:
  home: https://github.com/earthai-tech/hwm
  summary: Adaptive Hammerstein-Wiener Modeling Toolkit
  license: BSD-3-Clause
  license_file: LICENSE

extra:
  recipe-maintainers:
    - earthai-tech
    
```

**Comment by earthai-tech** [comment]
> As @ocefpaf mentioned, it's probably a good idea to look into using [`grayskull`](https://github.com/conda/grayskull) since it will fix a lot of the issues that are currently present in your PR since it reads directly from what's up on pypi.

I fix it

File: recipes/hwm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+
+requirements:
+  host:
+    - python >=3.9
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:20]:
```suggestion
    - python {{ python_min }}
```

File: recipes/hwm/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - pip
+    - setuptools
+  run:
+    - python >=3.9
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:24]:
```suggestion
    - python >={{ python_min }}
```

File: recipes/hwm/meta.yaml (hunk 2 above)
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:43]:
```suggestion
    - python {{ python_min }}
```

**Comment by earthai-tech** [comment]
All the fixes were done, I guess since all checks are passed. Please could you revise and merge?
@conda-forge/help-python

File: recipes/hwm/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  noarch: python
+
+requirements:
+  host:
+    - pip
+    - python >={{ python_min }}
+    - setuptools  
+
+  run:
+    - numpy <2
```
[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/hwm/meta.yaml:25]: With the exception of matplotlib, which has a lower bound upstream, so this is fine here, and joblib which 
https://github.com/earthai-tech/hwm/blob/v1.0.1/setup.py#L32-L35, which is not listed upstream. the rest seems fine.

File: recipes/hwm/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+test:
+  imports:
+    - hwm
+    - hwm.api
+    - hwm.compat
+    - hwm.datasets
+    - hwm.estimators
+    - hwm.utils
+  commands:
+    - python -c "import hwm; print(hwm.__version__)"
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:41]: What you have is a duplicate of the import test above. We need `pip check` here.

```suggestion
    - pip check
```

File: recipes/hwm/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  requires:
+    - python {{ python_min }}
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:43]:
```suggestion
    - python {{ python_min }}
    - pip
```

File: recipes/hwm/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    - pip
+    - setuptools
+  run:
+    - python >={{ python_min }}
+    - numpy <2
+    - scipy >=1.7.0
+    - scikit-learn >=1.2.0
+    - pandas >=1.3.0
+    - joblib >=1.2.0
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:29]: @earthai-tech I don't see joblib listed upstream. Is it really needed?

Comment (earthai-tech) [reply recipes/hwm/meta.yaml:29]: Not really needed for this version. I put it here for future version.

Comment (earthai-tech) [reply recipes/hwm/meta.yaml:29]: Ok. Got it! Thanks

**Review by earthai-tech — COMMENTED** [review]
Review is done. I removed joblib package

**Review by earthai-tech — COMMENTED** [review]
edit sha256  (65d481821c653baff10c617a5acc0e24c9b30ef1e4a61d211b65078725d98856) and move to version 1.14:

File: recipes/hwm/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/hwm-{{ version }}.tar.gz
+  sha256: 65d481821c653baff10c617a5acc0e24c9b30ef1e4a61d211b65078725d98856
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
```
Comment (ocefpaf) [inline recipes/hwm/meta.yaml:16]:
```suggestion
```

**Pull request merged**